var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -61,14 +61,24 @@ def mpi_dependent_options(self):
                 errors = ['incompatible variants given'] + errors
                 raise RuntimeError('\n'.join(errors))
         else:
-            compiler_opts = [
-                '--with-mpi=1',
-                '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
-            ]
+            if self.compiler.name == "clang":
+                compiler_opts = [
+                    '--with-mpi=1',
+                    '--with-cc=%s  -Qunused-arguments' % join_path(self.spec['mpi'].prefix.bin, 'mpicc'), # Avoid confusing PETSc config by clang: warning: argument unused during compilation
```

Comment:
In any case, you do want to add `-Qunused-arguments` to your wrappers as otherwise every build complains about `clang: warning: argument unused during compilation`.